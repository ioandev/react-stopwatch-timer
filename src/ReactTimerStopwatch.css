.react-stopwatch-timer__table{
  height: 100px;
  width: 100px;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.react-stopwatch-timer__container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}