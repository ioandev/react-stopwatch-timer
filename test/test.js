import React from 'react';
import ReactTimerStopwatch from '../src/ReactTimerStopwatch';
import renderer from 'react-test-renderer';

test('test react timer stopwatch component', () => {
    const component = renderer.create(
        <ReactTimerStopwatch isOn={true} className="react-stopwatch-timer__table" watchType="stopwatch"/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseEnter();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});