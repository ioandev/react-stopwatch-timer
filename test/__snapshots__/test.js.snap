// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test react timer stopwatch component 1`] = `
<div
  className="react-stopwatch-timer__table"
>
  <div>
    21:57:53
  </div>
</div>
`;
